resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -42,10 +44,22 @@ private[k8s] case class KubernetesDriverSpecificConf(
 private[k8s] case object KubernetesExecutorSpecificConf extends KubernetesRoleSpecificConf
 
 private[k8s] class KubernetesConf[T <: KubernetesRoleSpecificConf](
-    private val sparkConf: SparkConf,
-    val roleSpecificConf: T,
-    val appResourceNamePrefix: String,
-    val appId: String) {
+  private val sparkConf: SparkConf,
+  val roleSpecificConf: T,
+  val appResourceNamePrefix: String,
+  val appId: String,
+  val mainAppResource: Option[MainAppResource]) {
```

Comment:
Problem is that mainAppResource won't be available when creating one of these objects for Executor construction - hence it has to be in the role-specific conf. I think the caller can manually modify the SparkConf passed to `createDriverConf` to already have modified `spark.jars` accordingly.